src/gluonts/mx/block/quantile_output.py

```diff
@@ -183,103 +183,104 @@ def compute_quantile_weights(self) -> List:
+        return nn.Dense(units=self.num_quantiles, flatten=False)
 
-        Parameters
-        ----------
-        F
-            A module that can either refer to the Symbol API or the NDArray
-            API in MXNet.
-        x
-            input tensor
 
-        Returns
-        -------
-        Tensor
-            output of the projection layer
-        """
+class IncrementalDenseLayerProjection(nn.HybridBlock):
+    """
+    A dense layer that outputs non-decreasing values.
+
+    Parameters
+    ----------
+    num_outputs
+        number of outputs of the layer.
+    """
+
+    @validated()
+    def __init__(self, num_outputs: int, **kwargs):
+        super().__init__(**kwargs)
+
+        self.num_outputs = num_outputs
+        with self.name_scope():
+            self.proj_intrcpt = nn.Dense(1, flatten=False)
+            if self.num_outputs > 1:
+                self.proj_incrmnt = nn.Dense(
+                    self.num_outputs - 1,
+                    flatten=False,
+                    activation="relu",
+                )  # increments between quantile estimates
+
+    def hybrid_forward(self, F, x: Tensor) -> Tensor:
         return (
-            (
-                self.proj_intrcpt(x)
-                if self.num_quantiles == 1
-                else (
-                    F.cumsum(
-                        F.concat(
-                            self.proj_intrcpt(x), self.proj_incrmnt(x), dim=-1
-                        ),
-                        axis=3,
-                    )
+            self.proj_intrcpt(x)
+            if self.num_outputs == 1
+            else (
+                F.cumsum(
+                    F.concat(
+                        self.proj_intrcpt(x), self.proj_incrmnt(x), dim=-1
+                    ),
+                    axis=3,
                 )
             )
-            if self.is_iqf
-            else self.projection(x)
         )
 
 
-class QuantileOutput:
+class IncrementalQuantileOutput(QuantileOutput):
     """
     Output layer using a quantile loss and projection layer to connect the
     quantile output to the network.
 
+    The given quantile levels are sorted in ascending order, and the
+    projection layer will output the quantile values accordingly.
+
     Parameters
     ----------
     quantiles
         list of quantiles to compute loss over.
 
     quantile_weights
         weights of the quantiles.
-
-    is_iqf
-        determines whether to use IQF or QF.
     """
 
     @validated()
     def __init__(
         self,
         quantiles: List[float],
         quantile_weights: Optional[List[float]] = None,
-        is_iqf: bool = False,
     ) -> None:
-        self.quantiles = sorted(quantiles)
-        self.num_quantiles = len(self.quantiles)
-        self.quantile_weights = quantile_weights
-        self.is_iqf = is_iqf
-        self.is_equal_weights = False if self.is_iqf else True
+        super().__init__(sorted(quantiles), quantile_weights)
```

Comment:
ok so we still for the sorting here. I think the quantile knots for interpolation need to be sorted, but we can handle that inside of the `InterpolationClass` in `utils`

Reply:
Yes. Still I think it would be nice for the output of the projection (quantile values) to be in the originally intended order (the one specified by who constructed `IncrementalQuantileOutput`), and for that we could just reshuffle the quantiles back at the end of the projection.

Reply:
yes, I agree that we should return it in the specified order, and just internally sort inside of the `InterpolationClass` where it is required.